src/test/java/TasksServletTest.java

```diff
@@ -0,0 +1,152 @@
+          TASK_TITLE_ONE, TASK_TITLE_TWO, TASK_TITLE_THREE, TASK_TITLE_FOUR);
+  private static final String EMPTY_JSON = "[]";
+  private static final List<TaskList> noTaskLists = new ArrayList<>();
+  private static final List<TaskList> someTaskLists = Arrays.asList(new TaskList(), new TaskList());
+  private static final List<Task> noTasks = new ArrayList<>();
+  private static final List<Task> tasksOneTwo =
+      Arrays.asList(new Task().setTitle(TASK_TITLE_ONE), new Task().setTitle(TASK_TITLE_TWO));
+  private static final List<Task> tasksThreeFour =
+      Arrays.asList(new Task().setTitle(TASK_TITLE_THREE), new Task().setTitle(TASK_TITLE_FOUR));
+
+  @BeforeClass
+  public static void classInit() throws GeneralSecurityException, IOException {
+    when(tasksClientFactory.getTasksClient(Mockito.any())).thenReturn(tasksClient);
+    // Authentication will always pass
+    when(authenticationVerifier.verifyUserToken(Mockito.anyString()))
+        .thenReturn(AUTHENTICATION_VERIFIED);
+  }
+
+  @Before
+  public void init() throws IOException {
+    request = Mockito.mock(HttpServletRequest.class);
+    response = Mockito.mock(HttpServletResponse.class);
+    when(request.getCookies()).thenReturn(validCookies);
+
+    // Writer used in get/post requests to capture HTTP response values
+    stringWriter = new StringWriter();
+    printWriter = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(printWriter);
+  }
+
+  @After
+  public void clear() {
+    // Dump contents after each test
+    stringWriter.getBuffer().setLength(0);
+  }
+
+  @Test
+  public void noTaskLists() throws IOException {
+    when(tasksClient.listTaskLists()).thenReturn(noTaskLists);
+    servlet.doGet(request, response);
+    printWriter.flush();
+    Assert.assertTrue(stringWriter.toString().contains(EMPTY_JSON));
+  }
+
+  @Test
+  public void emptyTaskLists() throws IOException {
+    when(tasksClient.listTaskLists()).thenReturn(someTaskLists);
+    when(tasksClient.listTasks(Mockito.any())).thenReturn(noTasks);
+    servlet.doGet(request, response);
+    printWriter.flush();
+    Assert.assertTrue(stringWriter.toString().contains(EMPTY_JSON));
+  }
+
+  @Test
+  public void oneEmptyTaskList() throws IOException {
+    when(tasksClient.listTaskLists()).thenReturn(someTaskLists);
+    when(tasksClient.listTasks(Mockito.any())).thenReturn(noTasks).thenReturn(tasksOneTwo);
+    servlet.doGet(request, response);
+    printWriter.flush();
+    Assert.assertTrue(stringWriter.toString().contains(TASK_ONE_TWO_JSON));
+  }
+
+  @Test
+  public void completeTaskList() throws IOException {
+    when(tasksClient.listTaskLists()).thenReturn(someTaskLists);
+    when(tasksClient.listTasks(Mockito.any())).thenReturn(tasksOneTwo).thenReturn(tasksThreeFour);
+    servlet.doGet(request, response);
+    printWriter.flush();
+    Assert.assertTrue(stringWriter.toString().contains(TASK_ALL_JSON));
+  }
```

Comment:
You could add some comments to describe what is handled and what is expected

Reply:
I'm hesitant to add comments as they would explain the what not the why. The method names explain the case being tested and the code trivially explains the expected result